src/Pages/byungjinson/Login/Login.js

```diff
@@ -1,10 +1,91 @@
 import React, { Component } from "react";
+import { Link, withRouter } from "react-router-dom";
+import "./Login.scss";
+
 export class LoginSon extends Component {
+  constructor() {
+    super();
+    this.state = {
+      email: "",
+      password: "",
+    };
+  }
+  handleInput = (e) => {
+    const { name, value } = e.target;
+    this.setState({
+      [name]: value,
+    });
+  };
+
   render() {
     return (
-      <div>
+      <div className="loginSon">
+        <div className="login__input">
+          <img
+            id="instagramLogo"
+            alt="instagram_logo"
+            src="./Images/byungjinson/logo_text.png"
+          />
+          <div className="loginContainer" onChange={this.handleInput}>
+            <input
+              className="login__input__text"
+              name="email"
+              type="text"
+              placeholder="전화번호, 사용자 이름 또는 이메일"
+            />
+            <input
+              className="login__input__text"
+              name="password"
+              type="password"
+              placeholder="비밀번호"
+            />
+          </div>
+          <Link to="/mainSon">
+            <button
+              className={
+                this.state.email.includes("@") && this.state.password.length > 4
+                  ? "loginValid"
+                  : "loginUnvalid"
```

Comment:
굿입니다!
지금은 여기 한 곳에서 이 조건을 사용하고 있지만 만약에 다른 곳에서도 사용하게 된다면
이 조건 부분을 render와 return 사이에서 정의해주시고 사용해주세요! 그리고 비구조화도!
```suggestion
render(){
 const { email, password } = this.state;
 const isValid = email.includes("@") && password.length > 4;
  return(
  ...
               valid
                  ? "loginValid"
                  : "loginUnvalid"
  ...
  )
}
```

Reply:
오오 저 사이에도 변수 선언을 할 수 있군요.. 감사합니다
그리고 가운데 valid 아니라 isValid 가 맞는거죠..?